arrows/ffmpeg/ffmpeg_video_input.cxx

```diff
@@ -312,51 +310,16 @@ class ffmpeg_video_input::priv
   */
   void close()
   {
-    if (this->f_packet.data) {
-      av_free_packet(&this->f_packet);  // free last packet
-    }
-
-    if (this->f_frame)
-    {
-      av_frame_free(&this->f_frame);
-    }
-    this->f_frame = nullptr;
-    if (this->f_filtered_frame)
-    {
-      av_frame_free(&this->f_filtered_frame);
-    }
-    this->f_filtered_frame = nullptr;
-
-    if (this->f_video_encoding && this->f_video_encoding->opaque)
-    {
-      av_freep(&this->f_video_encoding->opaque);
-    }
-
     this->f_video_index = -1;
     this->f_data_index = -1;
     this->f_start_time = -1;
 
-    if (this->f_video_stream)
-    {
-      avcodec_close(this->f_video_stream->codec);
-      this->f_video_stream = nullptr;
-    }
-    if (this->f_format_context)
-    {
-      avformat_close_input(&this->f_format_context);
-      this->f_format_context = nullptr;
-    }
-    if (this->f_video_encoding)
-    {
-      avcodec_close(this->f_video_encoding);
-      avcodec_free_context(&this->f_video_encoding);
-      this->f_video_encoding = nullptr;
-    }
-    if (this->f_filter_graph)
-    {
-      avfilter_graph_free(&this->f_filter_graph);
-      this->f_filter_graph = nullptr;
-    }
+    av_frame_free(&this->f_frame);
```

Comment:
Noting that `f_video_stream` and `f_filter_graph` are no longer freed/closed.  Is this intentional?

Reply:
No - I added back `avfilter_graph_free(&this->f_filter_graph);` and the setting of `f_video-stream` to null. The `avcodec_close` call is deprecated. The codec is now held by `f_video_encoding`. The various `*_free` methods in ffmpeg handle the case when the pointer is null and set the pointer to null after freeing the memory so the protective if statements are not needed.